stdlib/public/Python/NumpyConversion.swift

```diff
@@ -152,41 +152,79 @@ extension Double : NumpyScalarCompatible {
   }
 }
 
-extension Array : ConvertibleFromNumpyArray
-  where Element : NumpyScalarCompatible {
+/// A helper struct with all the data that ConvertibleFromNumpyArray
+/// implementations need to copy memory from a NumpyArray.
+public struct ContiguousNumpyArray<Scalar, ShapeSize> {
+  /// The shape of the array.
+  public let shape: [ShapeSize]
```

Comment:
Why not use a `PythonObject` to represent shape? It's more efficient and consistent with the rest of the stored properties of the struct.

Reply:
sounds good -- it eliminates the problem of what type to give it